--- origsrc/audacity-minsrc-2.2.1-rc3/configure.ac	2018-01-11 20:38:42.040177400 -0600
+++ src/audacity-minsrc-2.2.1-rc3/configure.ac	2018-01-11 21:17:14.015027500 -0600
@@ -245,7 +245,7 @@ if "${CXX}" -v 2>&1 | grep -q '^gcc vers
         AC_MSG_ERROR([Audacity requires at least GCC 4.9])
     fi
 
-    CXXFLAGS="${CXXFLAGS} --std=gnu++11"
+    CXXFLAGS="${CXXFLAGS} --std=gnu++11 -D_GNU_SOURCE"
 else
     CXXFLAGS="${CXXFLAGS} --std=c++11"
 fi
